includes/core/server/request/class-list-transactions.php

```diff
@@ -0,0 +1,279 @@
+<?php
+/**
+ * Class file for WCPay\Core\Server\Request\List_Transactions.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use DateTime;
+use DateTimeZone;
+use WC_Payments_API_Client;
+use WC_Payments_DB;
+use WCPay\Core\Server\Response;
+use WP_REST_Request;
+
+/**
+ * Request class for listing transactions.
+ */
+class List_Transactions extends Paginated {
+
+	use Order_Info;
+
+	const DEFAULTS = [
+		'sort'       => 'date',
+		'direction'  => 'desc',
+		'deposit_id' => null,
+	];
+
+	/**
+	 * Set deposit id.
+	 *
+	 * @param mixed $deposit_id Deposit id.
+	 *
+	 * @return void
+	 */
+	public function set_deposit_id( $deposit_id ) {
+		$this->set_param( 'deposit_id', $deposit_id );
+	}
+
+	/**
+	 * Get api URI.
+	 *
+	 * @return string
+	 */
+	public function get_api(): string {
+		return WC_Payments_API_Client::TRANSACTIONS_API;
+	}
+
+	/**
+	 * Used to prepare request from WP Rest data.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return Paginated|self
+	 */
+	public static function from_rest_request( $request ) {
+		$wcpay_request       = parent::from_rest_request( $request );
+		$date_between_filter = $request->get_param( 'date_between' );
+		$user_timezone       = $request->get_param( 'user_timezone' );
+
+		if ( ! is_null( $date_between_filter ) ) {
+			$date_between_filter = array_map(
+				function ( $transaction_date ) use ( $user_timezone, $wcpay_request ) {
+					return $wcpay_request->format_transaction_date_with_timestamp( $transaction_date, $user_timezone );
```

Comment:
Even if we set the return type of `parent::from_rest_request()` to be `static`, it's still being returned as `Paginated`, which does not have `format_transaction_date_with_timestamp`, causing linting errors.

What do you think about making the method static as well? It does not need any object-specific functionality.

Reply:
Good point. Made this function static.